### PR TITLE
feat(gitlab): Update Integration to Prod

### PR DESCRIPTION
**Pull request by mbliakharskyi**
## Description

1. Implement user delete endpoint.
2. Update integration.
 - Removed /scripts folder
 - Updated the test with encrypted stuff
 - Renamed Organisation -> organisationsTable
 - Removed app.uninstalled event
 - Configured inngest functions

## Additional Notes

N/A

## Checklist:

Before you create this PR, confirm all the requirements listed below by checking the checkboxes like this (`[x]`).

- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have added tests to cover the new feature or fixes.

### CODE REVIEW

File: apps/gitlab/src/app/page.tsx [hunk 1]
```diff
@@ -0,0 +1,7 @@
+/**
+ * This file is required by nextjs and has no purpose for now in the integration.
+ * It should not be edited or removed.
+ */
+export default function Home() {
+  return <main>Elba x Box</main>;
```
Comment (elbalexandre) [inline apps/gitlab/src/app/page.tsx:6]:
```suggestion
  return <main>Elba x Gitlab</main>;
```

File: apps/gitlab/src/app/layout.tsx [hunk 2]
```diff
@@ -0,0 +1,18 @@
+/**
+ * This file is required by nextjs and has no purpose for now in the integration.
+ * It should not be edited or removed.
+ */
+import type { Metadata } from 'next';
+
+export const metadata: Metadata = {
+  title: 'Elba x Box',
```
Comment (elbalexandre) [inline apps/gitlab/src/app/layout.tsx:8]:
```suggestion
  title: 'Elba x Gitlab',
```

File: apps/gitlab/src/database/schema.ts [hunk 3]
```diff
@@ -0,0 +1,10 @@
+import { uuid, text, timestamp, pgTable } from 'drizzle-orm/pg-core';
+
+export const organisationsTable = pgTable('organisations', {
+  id: uuid('id').primaryKey(),
+  region: text('region').notNull(),
+  createdAt: timestamp('created_at').defaultNow().notNull(),
+  accessToken: text('accessToken').notNull(),
+  refreshToken: text('refreshToken').notNull(),
```
Comment (elbalexandre) [inline apps/gitlab/src/database/schema.ts:8]:
```suggestion
  accessToken: text('access_token').notNull(),
  refreshToken: text('refresh_token').notNull(),
```

File: apps/gitlab/src/inngest/functions/users/delete-users.ts [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+import { inngest } from '@/inngest/client';
+import { deleteUser } from '@/connectors/users';
+import { decrypt } from '@/common/crypto';
+
+export const deleteSourceUsers = inngest.createFunction(
+  {
+    id: 'gitlab-delete-users',
+    concurrency: {
+      key: 'event.data.organisationId',
+      limit: 5,
```
Comment (elbalexandre) [inline apps/gitlab/src/inngest/functions/users/delete-users.ts:14]: I just talk about it with @athielba we are going to use an env variable for this (with a default value of 5 it's fine) 

```suggestion
      limit: env.GITLAB_DELETE_USER_CONCURRENCY,
```

File: apps/gitlab/src/inngest/middlewares/rate-limit-middleware.ts [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+ * and it responds by creating a RetryAfterError. This error includes the retry time based on the 'Retry-After' header
+ * provided by the SaaS service, enabling the function to delay its next execution attempt accordingly.
+ *
+ * Key Features:
+ * - Intercepts function output to check for rate limit errors.
+ * - Handles GitlabError, specifically looking for a 'Retry-After' header in the error response.
+ * - Generates a RetryAfterError to reschedule the function run, preventing immediate retries that could violate the SaaS's rate limits.
+ *
+ * Note: This is a generic middleware template and might require adjustments to fit specific SaaS APIs' error handling and rate limiting schemes.
+ */
```
Comment (elbalexandre) [inline apps/gitlab/src/inngest/middlewares/rate-limit-middleware.ts:17]:
```suggestion
```

Could you add a test for this file ?

File: apps/gitlab/src/app/auth/route.ts [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+    return new ElbaInstallRedirectResponse({
+      region,
+      sourceId: env.ELBA_SOURCE_ID,
+      baseUrl: env.ELBA_REDIRECT_URL,
+      error: 'unauthorized',
+    });
+  }
+
+  await setupOrganisation({ organisationId, code, region });
+
```
Comment (elbalexandre) [inline apps/gitlab/src/app/auth/route.ts:26]: You can check how it's done in the template, there is an example with the logger.
```suggestion
try {
  await setupOrganisation({ organisationId, code, region });
} catch (error) {
  // log it
    return new ElbaInstallRedirectResponse({
      region,
      sourceId: env.ELBA_SOURCE_ID,
      baseUrl: env.ELBA_REDIRECT_URL,
      error: 'internal_error',
    });
}
```

File: apps/gitlab/src/inngest/middlewares/rate-limit-middleware.ts (hunk 5 above)
Comment (elbalexandre) [reply apps/gitlab/src/inngest/middlewares/rate-limit-middleware.ts:17]: There is surely one integration on staging that do a test for that, you could use it to implement your own.

Comment (mbliakharskyi) [reply apps/gitlab/src/inngest/middlewares/rate-limit-middleware.ts:17]: okay. will do

File: apps/gitlab/src/app/auth/route.ts (hunk 6 above)
Comment (mbliakharskyi) [reply apps/gitlab/src/app/auth/route.ts:26]: okay.

**Comment by elbalexandre** [comment]
I am closing this PR as deleting user requires us to store every group's users + project's users that we are owning.
We could this integration later.

**Pull request closed without merging**